test/system/reproducibleCompare/build.xml

```diff
@@ -14,32 +14,68 @@
   ~ ********************************************************************************
 -->
 
-<project name="Functional Reproducible Comparison Tests" default="build" basedir=".">
+<project name="System Reproducible Comparison Tests" default="build" basedir=".">
     <taskdef resource="net/sf/antcontrib/antlib.xml"/>
     <description>
         Temurin Reproducible Comparison tests
     </description>
-    <import file="${TEST_ROOT}/functional/build.xml"/>
-
+    <!-- <import file="${TEST_ROOT}/system/common.xml"/>  -->
     <!-- set global properties for this build -->
-    <property name="DEST" value="${BUILD_ROOT}/functional/reproducibleCompare"/>
-
+    <property name="DEST" value="${BUILD_ROOT}/system/reproducibleCompare"/>
+	<property environment="env"/>
     <target name="init">
         <mkdir dir="${DEST}"/>
     </target>
 
-    <target name="dist" description="generate the distribution">
-        <copy todir="${DEST}">
-            <fileset dir="./" includes="*.xml, *.mk, *.sh"/>
-        </copy>
-    </target>
-
-    <target name="clean" depends="dist" description="clean up">
-        <!-- Delete the ${build} directory trees -->
-        <delete dir="${build}"/>
-    </target>
-
-    <target name="build">
-        <antcall target="clean" inheritall="true"/>
-    </target>
-</project>
+	<target name="stage">
+		<if>
+			<isset property="env.VENDOR_TEST_REPOS"/>
+			<then>
+				<property name="buildRepoTemp" value="${env.VENDOR_TEST_REPOS}"/>
+			</then>
+			<else>
+				<property name="buildRepoTemp" value="https://github.com/adoptium/temurin-build.git"/>
+			</else>
+		</if>
+		<if>
+			<contains string="${SPEC}" substring="zos"/>
+			<then>
+				<propertyregex property="buildRepo" input="${buildRepoTemp}" regexp="https://github.com//" replace="git@github.com:" defaultValue="${buildRepoTemp}"/>
+			</then>
+			<else>
+				<property name="buildRepo" value="${buildRepoTemp}"/>
+			</else>
+		</if>
+		<if>
+			<isset property="env.VENDOR_TEST_BRANCHES"/>
+			<then>
+				<property name="buildBranch" value="${env.VENDOR_TEST_BRANCHES}"/>
+			</then>
+			<else>
+				<property name="buildBranch" value="master"/>
+			</else>
+		</if>
+		<exec executable="git" failonerror="true">
```

Comment:
is this cloning and checking out the exact correct version of temurin-build as the original JDK was built with?

Reply:
done by linux_repro_build.sh

Reply:
Reproduced JDK is using the parameters from SBOM file, which has nothing to do with this build step. This step ensures that files under temurin-build/test/system/reproducibleCompare and temurin-build/tooling/reproducible/ are using the specified repo and branch. Though if the specified repo and branch does change the file linux_repo_build_compare.sh the reproducing and  comparing process or rule may change. However it won't change the parameters from SBOM file.